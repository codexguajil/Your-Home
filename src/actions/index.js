export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS',
  projects
});