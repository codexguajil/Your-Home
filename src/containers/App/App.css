@import '~antd/dist/antd.css';


header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px;
}

h1 {
  font-family: 'Open Sans Condensed', sans-serif;
  font-size: 48px;
  text-align: center;
}

.fa-chevron-circle-right {
  font-size: 32px;
}
