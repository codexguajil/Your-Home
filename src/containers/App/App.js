import React, { Component } from 'react';
import NewProject from '../NewProject/NewProject';
import SideDrawer from '../SideDrawer/SideDrawer';
import ProjectsContainer from '../../components/ProjectsContainer/ProjectsContainer';
import { Link, Switch, Route } from 'react-router-dom';
import ProjectDetails from '../ProjectDetails/ProjectDetails';
import mockProjects from '../../mockProject';
import { fetchAllProjects } from '../../thunks/fetchAllProjects';
import { connect } from 'react-redux';

class App extends Component {
  componentDidMount = () => {
    this.props.fetchAllProjects();
  }

  renderProject = ({ match }) => {
    const project = mockProjects.data.projects.find(project => (project.id === parseInt(match.params.id)));
    return <ProjectDetails project={project} />
  }

  render() {
    return (
      <div className="App" id='App'>
        <header className="App-header">
          <SideDrawer />
          <Link to='/'>
            <h1>YourHome</h1>
          </Link>
          <NewProject />
        </header>
        <div>
          <Switch>
            <Route exact path='/'
              render={() => <ProjectsContainer projects={mockProjects.data.projects} />}
            />
            <Route exact path='/projects/:id'
              render={this.renderProject}
            />
          </Switch>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  fetchAllProjects: (projects) => dispatch(fetchAllProjects(projects))
});

export default connect(null, mapDispatchToProps)(App);