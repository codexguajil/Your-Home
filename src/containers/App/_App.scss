@import '~antd/dist/antd.css';

header {
  background: #3b4044;
}

h1 {
  font-family: 'Open Sans Condensed', sans-serif;
  font-size: 48px;
  text-align: center;
  color: #fff;
}

body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

