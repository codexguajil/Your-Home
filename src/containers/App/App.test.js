import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import { setProjects } from '../../actions';
import { mapStateToProps, mapDispatchToProps } from './App';

describe('App', () => {
  let wrapper;
  let mockFn;
  
  beforeEach(() => {
    mockFn = jest.fn();
    wrapper = shallow(<App fetchAllProjects={mockFn}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return a state object', () => {
    const mockState = {
      projects: [],
      materials: [],
      randomState: {}
    }
    const expected = {
      projects: []
    }

    const mockProps = mapStateToProps(mockState)
    expect(mockProps).toEqual(expected)
  });

  it.skip('should mapDispatchToProps', () => {
    const mockProjects = [{ name: 'mockProject1', id: '1', materials: [{}, {}]}, { name: 'mockProject2', id: '2', materials: [{}, {}]}, { name: 'mockProject3', id: '3', materials: [{}, {}]}]
    const mockDispatch = jest.fn()
    const actionToDispatch = setProjects(mockProjects)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.setProjects(mockProjects)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

});