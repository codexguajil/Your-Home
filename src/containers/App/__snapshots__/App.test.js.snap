// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
  id="App"
>
  <header
    className="App-header"
  >
    <SideDrawer />
    <Link
      to="/"
    >
      <h1>
        YourHome
      </h1>
    </Link>
    <ConnectFunction />
  </header>
  <div>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/projects/:id"
        render={[Function]}
      />
    </Switch>
  </div>
</div>
`;
